### ISSUE TITLE
libraw: 'auto_ptr' is deprecated

### ISSUE THREAD

**Issue by fabiencastan** [OpenImageIO/oiio#1990]
### Problem

I get the following error when building on CentOS 7 with Libraw 1.19.0:
```
/opt/AliceVision/build/external/libraw_build/include/libraw/libraw_datastream.h:122:33: error: 'auto_ptr' is deprecated (declared at /usr/include/c++/4.8.2/backward/auto_ptr.h:87) [-Werror=deprecated-declarations]
   std::auto_ptr<std::streambuf> saved_f; /* when *f is a subfile, *saved_f is the master file */
``` 
You can see the usage of auto_ptr [here](https://github.com/LibRaw/LibRaw/blob/master/libraw/libraw_datastream.h#L119).

**Expected behavior:**  

Would need to have:
```
    set_source_files_properties (rawinput.cpp
                             PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
```
As it is done on some other plugins, like ffmpeg [here](https://github.com/OpenImageIO/oiio/blob/e2eaa743d0563a6dd9c0b5b186a674749f38fb80/src/ffmpeg.imageio/CMakeLists.txt#L6)
I tried, but it was not taken into account... No idea why.

**Actual behavior:**  

I managed to get rid of it by adding: `-DCMAKE_C_FLAGS="-Wno-deprecated-declarations" ` to the build of openimageio.

### Steps to Reproduce

Build openimageio with LibRaw 1.19.0 on linux. I don't know if building on centos 7 is part of the game.
I don't know where the option `-Werror=deprecated-declarations` is added. This option is maybe not needed outside of the continuous integration.

### Versions

* OIIO branch/version: https://github.com/OpenImageIO/oiio/archive/Release-1.8.11.tar.gz
* OS: CentOS 7
* C++ compiler: g++ 4.8.5

**Comment by mfvescovi** [comment]
Hi!

LibRaw version is 0.19.0 (not 1.19.0) and it builds fine on Debian (see [here](https://github.com/OpenImageIO/oiio/files/2256531/openimageio_1.8.12.dfsg0-1.buildlog.gz)). This is a test build made for the upcoming transition of LibRaw 0.19.0 in Debian, superseding v.0.18.xx.
But, in Debian we use g++-8.2 actually, so probably the compiler is the culprit here, more than the OS itself.

Hope this helps.

**Comment by lgritz** [comment]
I believe this was fixed by #1974. Closing.
